bufferevent_filter.c

```diff
@@ -326,9 +327,22 @@ be_filter_process_input(struct bufferevent_filtered *bevf,
 		 evbuffer_get_length(bevf->underlying->input) &&
 		 !be_readbuf_full(bevf, state));
 
-	if (*processed_out)
+	if (*processed_out) {
 		BEV_RESET_GENERIC_READ_TIMEOUT(bev);
 
+		bufferevent_trigger_nolock_(bev, EV_READ, 0);
+		if (evbuffer_get_length(bevf->underlying->input) > 0 &&
+			be_readbuf_full(bevf, state)) {
+			/* data left in underlying buffer and filter input buffer
+			 * hit its read high watermark.
+			 * Schedule callback to avoid data gets stuck in underlying
+			 * input buffer.
+			 */
+			evbuffer_cb_set_flags(bev->input, bevf->inbuf_cb,
+				EVBUFFER_CB_ENABLED);
+		}
+	}
+
```

Comment:
I guess it make sense, can this be covered with unit test?

Reply:
I reverted this also for now for the reasons outlined in #613. It looks like we're not really using it either, though I may want to debug this more properly.

Reply:
I'm going to remove flushing out of this PR as it deserves a separate discussion